qiskit/transpiler/passes/optimization/template_matching/forward_match.py

```diff
@@ -404,6 +407,25 @@ def run_forward_match(self):
                 ):
                     continue
 
+                # Check if parameters match the template or not.
+                # Construct a temporary list of matches.
+                temp_match = self.match.copy()
+
+                # Append the new match to the list of matches.
+                temp_match.append([i, label])
+
+                # Check if the potential match is valid by attempting
+                # to bind parameters.
+                substitution = TemplateSubstitution(
+                    [temp_match], self.circuit_dag_dep, self.template_dag_dep, None
+                )
+
+                template_sublist, circuit_sublist = zip(*temp_match)
+                template = substitution._attempt_bind(template_sublist, circuit_sublist)
+
+                if template is None:
+                    continue
+
                 # Check if the qubit, clbit configuration are compatible for a match,
                 # also check if the operation are the same.
                 if (
```

Comment:
If I understand correctly, the original code was along the following lines:

```
match = False
for i in self.candidates:
    if match:
        break
    if (A):
        continue 
    if (B):
        ...
        match = True
        continue
```
and the fix checks that one additional condition has to be true for the match to happen 

```
match = False
for i in self.candidates:
    if match:
        break
    if (A):
        continue
    if (C):
        continue
    if (B):
        ...
        match = True
        continue
```  
where `C` is a somewhat expensive computation that uses `TemplateSubstitution._attempt_bind`.

Could you please try to rearrange the code as follows and see if this speeds things up a little bit:
```
match = False
for i in self.candidates:
    if match:
        break
    if (A):
        continue
    if (not B):
        continue
    if (C):
        continue
    
    ...
    match = True
    continue
```

Reply:
This is a minor point but maybe the code would be a bit more clear if the logic were as follows:
```
match = False
for i in self.candidates:
    if (A):
        continue
    if (not B):
        continue
    if (C):
        continue
    
    ...
    match = True
    break
```